src/post/workshed_utils.ts

```diff
@@ -50,26 +50,57 @@ const POTENTIAL_BEST_TRAIN_PIECES = [
       ),
   },
 ];
+function valueTrainStation(station: TrainSet.Station): number {
+  if (station === TrainSet.Station.COAL_HOPPER) {
+    return Math.max(...POTENTIAL_BEST_TRAIN_PIECES.map(({ value }) => value()));
+  }
+  return POTENTIAL_BEST_TRAIN_PIECES.find(({ piece }) => piece === station)?.value?.() ?? 0;
+}
 
-let trainCycle: TrainSet.Cycle;
-export function getBestCycle(): TrainSet.Cycle {
-  if (!trainCycle) {
-    const cycle = [
+let bestCycle: TrainSet.Cycle | null = null;
+export function getBestTrainConfiguration(): TrainSet.Cycle {
+  if (!bestCycle) {
+    bestCycle = [
       TrainSet.Station.COAL_HOPPER,
       ...POTENTIAL_BEST_TRAIN_PIECES.sort(({ value: a }, { value: b }) => b() - a()).map(
         ({ piece }) => piece
       ),
       TrainSet.Station.TOWER_FIZZY,
       TrainSet.Station.VIEWING_PLATFORM,
     ] as TrainSet.Cycle;
-    trainCycle = cycle;
   }
-  return [...trainCycle];
+  return bestCycle;
+}
+
+let bestStations: TrainSet.Station[] | null = null;
+export function getBestTrainStations(): TrainSet.Station[] {
+  if (!bestStations) {
+    const cycle = getBestTrainConfiguration();
+    const cycleValue =
+      5 * (sum(POTENTIAL_BEST_TRAIN_PIECES, ({ value }) => value()) + valueTrainStation(cycle[1]));
+
+    const goodStations: TrainSet.Station[] = cycle.slice(0, 2);
+    let currentBestAverageTurnValue = (cycleValue + sum(goodStations, valueTrainStation)) / 42;
+
+    for (let i = 2; i < 7; i++) {
+      const averageTurnValue =
+        (cycleValue + sum([...goodStations, cycle[i]], valueTrainStation)) / (40 + i);
+      if (averageTurnValue > currentBestAverageTurnValue) {
+        currentBestAverageTurnValue = averageTurnValue;
+        goodStations.push(cycle[i]);
+      } else {
+        bestStations = goodStations;
+        break;
+      }
```

Comment:
`const goodStations = cycle.slice(0, [...Array(cycle.length-1).keys()].map((i) => (cycleValue + sum(cycle.slice(0, i+2, valueTrainStation))) / (42+i)).map((x, i) => [x, i+2]).reduce((r, a) => (a[0] > r[0] ? a : r))[1] ?? 2)`

Reply:
Given that we're caching this and it's a very small array, I think we can call both of these versions "extremely cheap"

I think I will probably personally abandon this PR and let you take it over slash do it in another PR. That being said, at least one of those reduces looks like it should be a `flat`, and the other map + reduce looks like it should maybe be a `maxBy`